content/news/buzz/2025q1-roadmap/contents.lr

```diff
@@ -0,0 +1,44 @@
+
+Q1 priorities
+--------------
+
+In Q1, we will continue to work on binary packaging for mobile platforms. We will continue to work on landing the patches to ``cibuildwheel`` and Pillow for iOS support; once that work has concluded, we hope to start making the analogous changes required to support Android.
+
+Q1 will also see the continuation of our collaboration with PyScript. The major focus for BeeWare in this work will be Toga's Web backend - and in particular, Toga's testing infrastructure. We're not currently able to run Toga's testbed project on the web backend (or the Textual backend, for that matter). We have some ideas for how this could be addressed; these ideas may not be fully realised by the end of Q1, but we hope to at least have a concrete plan, and maybe even a prototype in place. 
+
+Lastly, with the CPython 3.14 development process well underway, there are two projects that we intend to work on, with the goal of having this work incorporated into Python 3.14. The first is the creation of official "embedded" distributions of CPython for iOS, Android and macOS. Python already produces an embedded build for Windows, and Briefcase is able to use this distribution as part of the process of creating Windows binaries. We hope that Python 3.13 will contain analogous builds for iOS and Android, removing the need for BeeWare (and other projects) to maintain independent CPython builds. We've had some initial discussions with the CPython release team; we hope that by the end of the quarter, we'll have a plan for how to deliver this work. 
+
+The second CPython feature we'd like to address is handling of cross-platform virtual environments - that is, creating a virtual environment that can "pretend" it is running on a different host platform. This is necessary for cross-platform builds, as creating an isolated PEP 517 build environment requires the ability to create an environment that can run Python on the local machine, but report platform-identifying information as if it were running on the target machine. A version of this cross-environment virtualization has been necessary to support ``cibuildwheel``; however, it would be desirable to upstream this work so that it can be standardised as part of a stock Python environment. We hope to start discussions about this with the CPython core team in the near future.
```

Comment:
```suggestion
The second CPython feature we'd like to address is handling of cross-platform virtual environments - that is, creating a virtual environment that can "pretend" it is running on a different host platform. This is necessary for cross-platform builds, which require the ability to create an isolated PEP 517 build environment that can run Python on the local machine, but report platform-identifying information as if it were running on the target machine. A version of this cross-environment virtualization has been necessary to support ``cibuildwheel``; however, it would be desirable to upstream this work so that it can be standardised as part of a stock Python environment. We hope to start discussions about this with the CPython core team in the near future.
```